include/swift/ABI/ObjectFile.h

```diff
@@ -0,0 +1,99 @@
+//===--- ObjectFile.h - Object File Related Information ------*- C++ -*-===//
+//
+// Object File related data structures.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_ABI_OBJECTFILE_H
+#define SWIFT_ABI_OBJECTFILE_H
+
+#include "llvm/Support/ErrorHandling.h"
+#include "llvm/ADT/StringRef.h"
+
+namespace swift {
+
+/// Represents the six reflection sections used by Swift
+enum ReflectionSectionType : uint8_t {
```

Comment:
I'm sorry for requesting yet another change, but would you mind calling this
ReflectionSectionKind instead? That's how we usually call discriminators.

Reply:
No problem!